pkg/injection/namespace/mapper/mapper.go

```diff
@@ -175,3 +176,34 @@ func isIgnoredNamespace(dk *dynakube.DynaKube, namespaceName string) bool {
 
 	return false
 }
+
+func ListMonitoredNamespaces(ctx context.Context, clt client.Reader, dk *dynakube.DynaKube) (oneAgent []string, metaEnrich []string, _ error) {
```

Comment:
so, here we are kind doing "double work", or introducing it atleast.

We already request the namespaces: https://github.com/Dynatrace/dynatrace-operator/blob/b57c523e610621b5cdef7cc4538ebbf98e6b57a7/pkg/injection/namespace/mapper/dynakubes.go#L48
- (as part of the mapping)

we also do this matching here https://github.com/Dynatrace/dynatrace-operator/blob/b57c523e610621b5cdef7cc4538ebbf98e6b57a7/pkg/injection/namespace/mapper/mapper.go#L66

SO, IMO we can either:
a) improve the mapper so we do not do double the work
b) do it how it is done now, because the "long term" plan is to remove this `mapper` because we don't want to need have namespace webhook and namespace labeling (due to permissions)